### PR TITLE
Homework2:Pavel Martysiuk

### PR DESCRIPTION
**Pull request by PavelMartysiuk**

### CODE REVIEW

File: src/task3.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+def fizz_bazz(range_):
+    for digit in range(range_):
+        if digit % 3 == 0 and digit % 5 == 0:
```
Comment (belovmd) [inline src/task3.py]: you can use 15 here

File: src/task1.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+def money_calculator(cost, quantity):
```
Comment (belovmd) [inline src/task1.py]: please solve it according to task description. 
input: M, N, L etc

**Review by belovmd — CHANGES_REQUESTED** [review]
1. please remove tasks from first HW
2. please put tasks descriptions to docstrings
3. please add palindrom task
4. please address next issues:

File: src/first_word.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'''Print first word in string'''
+
+
+def first_word(str_):
+    word = ''
+    for sym in str_:
+        if sym == ' ':
```
Comment (belovmd) [inline src/first_word.py:7]: it wouldn't work with `Hello, world'

File: src/isogram.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+''' Write program that check str is isogram.'''
```
Comment (belovmd) [inline src/isogram.py:1]: no description what is isogram in this tasks (there are some options for in)

File: src/isogram.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def iso(str_):
+    str_ = str_.lower()
+    dicty = {}
+    for sym in str_:
```
Comment (belovmd) [inline src/isogram.py:7]: it's complexity is O(n**2) - too complex (count inside for)

File: src/isogram_2.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+'''Isogram sedond version'''
+
+
+def iso(str_):
+    if len(str_) == len(set(str_.lower())):
```
Comment (belovmd) [inline src/isogram_2.py:5]: `return len(str_) == len(set(str_.lower()))` would be enough

File: src/max_cost.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+'''Find max value in dict'''
+
+
+def max_cost(dict_):
+    return max(dict_.items(), key=lambda x: x[-1])
```
Comment (belovmd) [inline src/max_cost.py:5]: do we really need this lambda?

File: src/power.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
+'''raise to the power args'''
```
Comment (belovmd) [inline src/power.py:1]: 1. Unclear description
2. please use tripple double quotes

File: src/task1.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+'''Purchase money calculator'''
```
Comment (belovmd) [inline src/task1.py:1]: """ please put docstring """
unspecified input

File: src/task2.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+'''Palindrome'''
```
Comment (belovmd) [inline src/task2.py:1]: """

File: src/task2.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def palindrome(num):
```
Comment (belovmd) [inline src/task2.py:4]: unspecified input

File: src/task2.py [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+    power = len(str(num)) - 1
+    for i in range(1, (len(str(num)) // 2) + 1):
+        first_digit = num // 10 ** power
```
Comment (belovmd) [inline src/task2.py:7]: if we just revert num it would be much more clear

File: src/task3.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
+'''FizzBazz'''
```
Comment (belovmd) [inline src/task3.py:1]: """FixxBUzz"""

File: src/task3.py [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def fizz_bazz(range_):
+    for digit in range(range_):
```
Comment (belovmd) [inline src/task3.py:5]: it would pring 0-99 instead of 100

**Review by belovmd — COMMENTED** [review]
1. why do you need to move \_\_init__ ?

7.5

File: src/max_cost.py (hunk 7 above)
Comment (PavelMartysiuk) [reply src/max_cost.py]: Yes, if  I don't use lambda program return a max key.

Comment (PavelMartysiuk) [reply src/max_cost.py:5]: Yes, if I don't use the lambda function, the function return a max key of dict.

**Pull request closed without merging**